www/src/components/accordion.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useMemo, useContext, useRef, useEffect } from "react"
+import { colors, transition, space, radii } from "../utils/presets"
+import ChevronSvg from "./sidebar/chevron-svg"
+
+const AccordionContext = React.createContext()
+
+export const Accordion = ({ children }) => {
+  const [childState, setChildState] = useState([])
+  const lastRegisteredId = useRef(0)
+  const accordionContextValue = useMemo(() => {
+    return {
+      registerChild: ref => {
+        ref.current = ++lastRegisteredId.current
+        setChildState(previousState => {
+          const newState = [...previousState]
+          newState[ref.current] = false
+          return newState
+        })
+      },
+      getExpandedState: id => !!childState[id],
+      toggleExpandedState: id =>
+        setChildState(previousState =>
+          previousState.map((isOpen, index) => {
+            if (index === id) {
+              return !isOpen
+            } else {
+              return false
+            }
+          })
+        ),
+    }
+  }, [childState])
+
+  return (
+    <AccordionContext.Provider value={accordionContextValue}>
+      {children}
+    </AccordionContext.Provider>
+  )
+}
+
+const useAccordionState = () => {
+  const accordionContext = useContext(AccordionContext)
+  const id = useRef()
+  useEffect(() => {
+    if (accordionContext && !id.current) {
+      accordionContext.registerChild(id)
+    }
+  })
+  if (accordionContext && id.current) {
+    return [
+      accordionContext.getExpandedState(id.current),
+      e => {
+        e.preventDefault()
+        accordionContext.toggleExpandedState(id.current)
+      },
+    ]
+  } else {
+    return [null, null]
+  }
+}
+
+const DetailsContext = React.createContext({
+  toggle: () => void 0,
+})
+
+export const Details = props => {
+  const [open, toggle] = useAccordionState()
+  const contextValue = useMemo(() => {
+    return {
+      toggle,
+    }
+  }, [toggle])
+  return (
+    <DetailsContext.Provider value={contextValue}>
+      <details open={open} {...props} />
```

Comment:
The details and summary elements are great, but they aren't supported everywhere. For browsers that don't support it (such as Edge, until users upgrade to the Chromium-based version), this is how it renders:

![details collapsed with summaries showing](https://user-images.githubusercontent.com/1045233/65633322-aa7b4200-df90-11e9-9f45-c508caef098e.png)

Can we shim support somehow? Here's an article on it: https://www.smashingmagazine.com/2014/11/complete-polyfill-html5-details-element/

Reply:
We could easily shim support, yeah. I was worried about accessibility. since any shim would have to have `role="button"` but that would hide the heading inside the `summary` element from the a11y tree because buttons can't have headings inside. 

What would you prefer we lose in legacy browsers, a bit of screen-reader usability or the ability to expand and collapse?

I'll fix the styling so the spacing is less awkward in either case. I might need some pointers on getting Gatsby building in IE11 compatibility mode the default develop server didn't load at all when I tried it.